### PR TITLE
New tile-coloring turf decals!

### PR DESCRIPTION
**Pull request by Greniza**
## About The Pull Request

Adds new turf decals for the new tileset.

## Why It's Good For The Game

This is all that is capable with the old system with tiles:
image

All those colorless cracks in the paint! Ick! We should be able to do better. Now we can!
image

## Changelog
:cl:
imageadd: New turf decals for alternative tiletypes
/:cl:

### CODE REVIEW

File: code/game/objects/effects/decals/turfdecal/tilecoloring.dm [hunk 1]
```diff
@@ -21,52 +21,283 @@
 
 #undef PRIDE_ALPHA
 
+/// Blue tiles
+
 /obj/effect/turf_decal/tile/blue
 	name = "blue corner"
+	var/color_name = "blue"
```
Comment (Fikou) [inline code/game/objects/effects/decals/turfdecal/tilecoloring.dm:28]: wtf are u doing

File: code/game/objects/effects/decals/turfdecal/tilecoloring.dm [hunk 2]
```diff
@@ -21,52 +21,283 @@
...
+
+/obj/effect/turf_decal/tile/random/full
+	icon_state = "tile_full"
+	name = "colorful full"
+
 /obj/effect/turf_decal/tile/random/Initialize()
 	color = "#[random_short_color()]"
 	. = ..()
 
+/// Done with tile decals
```
Comment (Fikou) [inline code/game/objects/effects/decals/turfdecal/tilecoloring.dm:300]: why is this an autodoc comment

**Comment by Fikou** [comment]
the tile joiners of all sorts are really overcomplicating stuff imo, you should keep it to the half, anticorner and full decals

File: code/game/objects/effects/decals/turfdecal/tilecoloring.dm (hunk 1 above)
Comment (Greniza) [reply code/game/objects/effects/decals/turfdecal/tilecoloring.dm:28]: Thing from before second commit. Forgot to take it out, it was late.

**Comment by Greniza** [comment]
> the tile joiners of all sorts are really overcomplicating stuff imo, you should keep it to the half, anticorner and full decals

The idea is flexibility, so that you can do whatever you want with any of the tiles. I see your perspective, but as I've already written stuff for them, I see no need to unimplement it; people could just not use it.

**Comment by Fikou** [comment]
it clogs up the decal list which is already hard to sort through with stuff that wont be used 90% of the time

**Comment by Greniza** [comment]
> it clogs up the decal list which is already hard to sort through with stuff that wont be used 90% of the time

Everything's nested below the corresponding tile corner. Unless you have the entire decal tree open, it shouldn't.

**Review by Twaticus — CHANGES_REQUESTED** [review]
agreed with fikou, i dont think the joint pieces are necessary

**Comment by Greniza** [comment]
Literally 1984

**Review by Twaticus — APPROVED** [review]

**Pull request merged**